src/hooks/useMMKV.ts

```diff
@@ -154,8 +171,10 @@ export const useMMKVStorage = <T>(
     [key, storage, valueType]
   );
 
-  defaultValue = defaultValue === undefined ? null : defaultValue;
-  return [valueType === 'boolean' ? value : value || defaultValue, setNewValue];
+  return [
+    valueType === 'boolean' ? value : value || (defaultValue === undefined ? null : defaultValue),
+    setNewValue
+  ];
```

Comment:
After mimicking the behavior of `useState` we should remove `defaultValue === undefined` check and not decide if the value should be null or undefined unless the default value is explicitly set to `null`.

Something like this.
```js
return [
    valueType === 'boolean' || valueType === 'number' ? value : value || defaultValue,
    setNewValue
  ];
```
Numbers and booleans can have falsy values that are still required.

Reply:
I'll remove the `defaultValue === undefined`, but regarding the falsy values, I solved it on the following commit [eddf740](https://github.com/ammarahm-ed/react-native-mmkv-storage/pull/254/commits/eddf740083ccbbd66296388b924fd876355bf529) using the [nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator), was that not valid?

Reply:
@douugbr I think we should not decide for cases other than numbers (0, -1) & boolean (false) values. It's responsibility of the app/dev to pass a valid value, so if it's an empty string, it's an empty string. Also i think overriding the `defaultValue` passed will cause more inconsistencies and confusion so we should do it only when necessary.